Flub/reset rtt

flub opened the pull request:
Part of https://github.com/n0-computer/iroh/pull/2234

flub committed:
Hack in resetting of congestion controller

This is a super hacky way to be able to reset the right bits of the
congestion controller.

flub committed:
Some saner version of this

flub committed:
allow access to the connection inner

dignifiedquire commented:
@flub given the actual diff is not that large, maybe we should open a Draft PR to `quinn` directly to get some feedback/see if there is a chance we can avoid running a fork for too long

flub commented:
@dignifiedquire Yep, we should try, though in the current form I guess it's unlikely because I'm explicitly making internal things public.

flub committed:
change how the rtt is reset

There was too much state in the path that was being thrown away.
Possibly there is not enough being reset right now, especially the mtu
might matter.  But it at least works.
